libcnb-cargo/src/main.rs

```diff
@@ -0,0 +1,218 @@
+        Ok(binary_path) => binary_path,
+        Err(error) => {
+            error!("Packaging buildpack failed due to a build related error!");
+
+            match error {
+                BuildError::IoError(io_error) => {
+                    error!("IO error while executing Cargo: {}", io_error);
+                }
+                BuildError::UnexpectedExitStatus(exit_status) => {
+                    error!(
+                        "Unexpected Cargo exit status: {}",
+                        exit_status
+                            .code()
+                            .map_or_else(|| String::from("<unknown>"), |code| code.to_string())
+                    );
+                    error!("Examine Cargo output for details and potential compilation errors.");
+                }
+                BuildError::CrossCompileError(_) => {
+                    error!(
+                        "Could not find required linker and C compiler for the target platform!"
+                    );
+                    if let Some(help_text) = cross_compile_help(&target_triple) {
+                        error!("Hint:\n{}", help_text);
+                    }
+                }
+                BuildError::NoTargetsFound => {
+                    error!("No targets were found in the Cargo manifest. Ensure that there is exactly one binary target and try again.");
+                }
+                BuildError::MultipleTargetsFound => {
+                    error!("Multiple targets were found in the Cargo manifest. Ensure that there is exactly one binary target and try again.");
+                }
+                BuildError::MetadataError(metadata_error) => {
+                    error!("Unable to obtain metadata from Cargo: {}", metadata_error);
+                }
+                BuildError::CouldNotFindRootPackage => {
+                    error!("Root package could not be determined from the Cargo manifest.");
+                }
+            }
+
+            std::process::exit(1);
+        }
+    };
+
+    info!("Writing buildpack tarball...");
+    if let Err(io_error) = assemble_buildpack_tarball(
+        &output_path,
+        &buildpack_data.buildpack_toml_path,
+        &binary_path,
+    ) {
+        error!("IO error while writing buildpack tarball: {}", io_error);
+        std::process::exit(1);
+    };
+
+    info!(
+        "Successfully wrote buildpack tarball: {} ({})",
+        relative_output_path.to_string_lossy(),
+        output_path.metadata().map_or_else(
+            |_| String::from("unknown size"),
+            |metadata| SizeFormatterSI::new(metadata.len()).to_string()
+        )
+    );
+
+    info!("Packaging successfully finished!");
+    info!("Hint: To test your buildpack locally with pack, run: pack build my-image -b {} --path /path/to/application", relative_output_path.to_string_lossy());
+}
+
+fn setup_cli_parsing<'a, 'b>() -> clap::App<'a, 'b> {
```

Comment:
I'd like to see this moved into it's own file. This is a pretty big `main.rs`. From various CLIs that I've been exposed to in rust will have a [`args.rs`](https://github.com/BurntSushi/ripgrep/blob/master/crates/core/main.rs#L56) or `cli.rs` that encapsulates this.

Reply:
I'd argue that 200 lines is hardly "big" and honestly I'm totally fine leaving that function in `main.rs`. But to not enter a bikeshed loop, I've moved this out in e1b8090.